### PR TITLE
NMS-16150: Add docs on TL1d

### PR DESCRIPTION
**Pull request by mmahacek**
### All Contributors

* [X] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [X] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-16150

### CODE REVIEW

File: docs/modules/operation/pages/deep-dive/events/sources/tl1.adoc [hunk 1]
```diff
@@ -4,6 +4,115 @@
...
+| n/a
+
+3+|*Optional*
+
+| port
+| The TCP port on the node identified by the host attribute on which Tl1d will connect.
+| 502
+
+| userid
+| A user ID with which to authenticate to the element
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/events/sources/tl1.adoc:60]:
```suggestion
| A user ID with which to authenticate to the element.
```

File: docs/modules/operation/pages/deep-dive/events/sources/tl1.adoc [hunk 2]
```diff
@@ -4,6 +4,115 @@
...
+| password
+| The password corresponding to the user identified by the userid attribute.
+| opennms
+
+| reconnect-delay
+| The time (in milliseconds) that Tl1d should wait before reconnecting to this element if it becomes disconnected.
+| 30000
+
+| tl1-client-api^1^
+| The fully-qualified Java class name of a class that provides methods for communicating with this element.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/events/sources/tl1.adoc:72]:
```suggestion
| The fully qualified Java class name of a class that provides methods for communicating with this element.
```

File: docs/modules/operation/pages/deep-dive/events/sources/tl1.adoc [hunk 3]
```diff
@@ -4,6 +4,115 @@
...
+| org.opennms.netmgt.tl1d.Tl1ClientImpl
+
+| tl1-message-parser^1^
+| The fully-qualified Java class name of a class that provides methods for parsing messages received from this element.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/events/sources/tl1.adoc:76]:
```suggestion
| The fully qualified Java class name of a class that provides methods for parsing messages received from this element.
```

File: docs/modules/operation/pages/deep-dive/events/sources/tl1.adoc [hunk 4]
```diff
@@ -4,6 +4,115 @@
...
+| org.opennms.netmgt.tl1d.Tl1AutonomousMessageProcessor
+|===
+^1^ These two attributes should rarely, if ever, need to be changed from their defaults.
+
+== Event format
+
+Tl1d automatically subscribes to and recognizes TL1 autonomous messages received from elements to which it has successfully connected and authenticated.
+It parses these autonomous messages and creates {page-component-title} events.
+These events have a UEI of `uei.opennms.org/api/tl1d/message/autonomous`, and convey information from the autonomous message with following event parameters:
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/events/sources/tl1.adoc:85]:
```suggestion
These events have a UEI of `uei.opennms.org/api/tl1d/message/autonomous`, and convey information from the autonomous message with the following event parameters:
```

File: docs/modules/operation/pages/deep-dive/events/sources/tl1.adoc [hunk 5]
```diff
@@ -4,6 +4,115 @@
...
+
+raw-message:: The original, unprocessed TL1 autonomous message.
+alarm-code:: The alarm-code field from the received TL1 autonomous message.
+atag:: The Autonomous Tag (ATAG) field from the received TL1 autonomous message.
+verb:: The Verb field from the received TL1 autonomous message, one of REPT (report), EVT (event), or SESSION.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/events/sources/tl1.adoc:90]:
```suggestion
verb:: The Verb field from the received TL1 autonomous message; one of REPT (report), EVT (event), or SESSION.
```

File: docs/modules/operation/pages/deep-dive/events/sources/tl1.adoc [hunk 6]
```diff
@@ -4,6 +4,115 @@
...
+autoblock:: The Auto Block field from the received TL1 autonomous message.
+aid:: The Access Identifier (AID) field from the received TL1 autonomous message.
+additionalParams:: Any parameters from the Auto Block of the received TL1 autonomous message beyond AID, notification code (if present), and severity.
+
+== Handling events
+
+As all tl1d events come in with a uniform UEI, here are some ways you could handle the received events.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/events/sources/tl1.adoc:97]:
```suggestion
As all Tl1d events come in with a uniform UEI, here are some ways you could handle the received events.
```

File: docs/modules/operation/pages/deep-dive/events/sources/tl1.adoc [hunk 7]
```diff
@@ -4,6 +4,115 @@
...
+
+* Event translation and passive status
++
+A common use case might be to represent individual TL1 elements, which are typically not reachable directly via IP, as passive nodes in {page-component-title}.
+The xref:operation:deep-dive/events/event-translator.adoc[Event Translator] and xref:reference:service-assurance/monitors/PassiveStatusMonitor.adoc[Passive Status Keeper] features allow the state of various services to be updated through a specifically crafted event.
+
+* Alarms, automations, and trouble ticketing
++
+The default TL1 autonomous message event in {page-component-title} is annotated with alarm data.
+By customizing that alarm data to your particular needs, you can enable sophisticated automations that perform event enrichment and integrate with external systems.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/events/sources/tl1.adoc:107]:
```suggestion
By customizing that alarm data to your needs, you can enable sophisticated automations that perform event enrichment and integrate with external systems.
```

File: docs/modules/reference/pages/daemons/daemon-config-files/tl1d.adoc [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+[options="header"]
+[cols="2,3,1,1"]
+|===
+| File
+| Description
+| Reload Event
+| Restart Required
+
+| tl1d-configuration.xml
+| Configure generic behavior of tl1d.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/daemons/daemon-config-files/tl1d.adoc:18]:
```suggestion
| Configure generic behavior of Tl1d.
```

File: docs/modules/reference/pages/daemons/daemon-config-files/tl1d.adoc [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+.TL1d log file overview
+[options="header"]
+[cols="2,3"]
+|===
+| File
+| Description
+
+| tl1d.log
+| Logs relating to tl1 message parsing.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/daemons/daemon-config-files/tl1d.adoc:31]:
```suggestion
| Logs relating to Tl1 message parsing.
```

**Review by Bonrob2 — APPROVED** [review]
LGTM! Thanks!

**Review by garcia-sr — APPROVED** [review]
LGTM!

**Pull request merged**